### DISCUSSION TITLE
go build fail

### DISCUSSION THREAD

**Discussion by SunnyDate** [go-gorm/gorm#6538] [Q&A]
go version
go version go1.17 linux/amd64

go build
# gorm.io/gorm
./gorm.go:382:64: reflect.ValueOf(tx).Elem().FieldByName("db").UnsafePointer undefined (type reflect.Value has no field or method UnsafePointer)

**Comment by maranqz** [comment]
UnsafePointer was added in go 1.18. https://pkg.go.dev/reflect@go1.18.10#Value.UnsafePointer

In go 1.16 and 1.17, there is not UnsafePointer. https://pkg.go.dev/reflect@go1.16.15#Value.UnsafePointer

Problem was happened after https://github.com/go-gorm/gorm/pull/6535
